src/main/java/org/broadinstitute/hellbender/engine/ReadWalker.java

```diff
@@ -98,15 +105,50 @@ private SimpleInterval getReadInterval(final GATKRead read) {
 
     /**
      * Returns the read filter (simple or composite) that will be applied to the reads before calling {@link #apply}.
-     * The default implementation uses the {@link WellformedReadFilter} filter with all default options.
-     * Default implementation of {@link #traverse()} calls this method once before iterating
-     * over the reads and reuses the filter object to avoid object allocation. Nevertheless, keeping state in filter objects is strongly discouraged.
+     * The default implementation combines the default read filters for this tool (returned by
+     * {@link org.broadinstitute.hellbender.engine.ReadWalker#getDefaultReadFilters} with any read filter command
+     * line arguments specified by the user; wraps each filter in the resulting list with a CountingReadFilter;
+     * and returns a single composite filter resulting from the list by and'ing them together.
      *
-     * Subclasses can extend to provide own filters (ie override and call super).
-     * Multiple filters can be composed by using {@link org.broadinstitute.hellbender.engine.filters.ReadFilter} composition methods.
+     * Default tool implementation of {@link #traverse()} calls this method once before iterating
+     * over the reads and reuses the filter object to avoid object allocation. Nevertheless, keeping state in filter
+     * objects is strongly discouraged.
+     *
+     * Multiple filters can be composed by using {@link org.broadinstitute.hellbender.engine.filters.ReadFilter}
+     * composition methods.
      */
     public CountingReadFilter makeReadFilter(){
-          return new CountingReadFilter("Wellformed", new WellformedReadFilter(getHeaderForReads()));
+        // This filter gets returned if all others are disabled.
+        CountingReadFilter defaultFilter = new CountingReadFilter(
+                ReadFilterLibrary.ALLOW_ALL_READS.getClass().getSimpleName(),
+                ReadFilterLibrary.ALLOW_ALL_READS);
+
+        GATKReadFilterPluginDescriptor readFilterPlugin =
+                (GATKReadFilterPluginDescriptor)
+                        commandLineParser.getPluginDescriptor(GATKReadFilterPluginDescriptor.class);
+
+        // Unless all filters are disabled, merge the tool's default filters with the users's command line
+        // read filter requests, then initialize the resulting filters and wrap each one in a CountingReadFilter.
+        return readArguments.disableAllReadFilters == true ?
```

Comment:
`disableAllReadFilters` should be handled internally in the descriptor validation